src/index.ts

```diff
@@ -161,40 +165,30 @@ _getAllByOS(options, callback) {
 
   var osParts = this._parseOsInput(options.osNames[0]);
 
-  var reqOpts: any = {
-    uri: osParts.url,
-    json: true,
-    query: {},
+  var reqOpts: AxiosRequestConfig = {
+    url: osParts.url,
+    params: {}
   };
 
   if (osParts.version.length > 0) {
-    reqOpts.query.filter =
+    reqOpts.params.filter =
       'name eq ' + [osParts.name, osParts.version].join('-') + '.*';
   }
 
-  this._auth.authorizeRequest(reqOpts, function(err, authorizedReqOpts) {
-    if (err) {
-      callback(err);
```

Comment:
(I have no idea what this code does)
...but it's a change in behavior, isn't it? no need to call `callback(err)` in case if `.request()` failed?

Reply:
This is a great question! A lot of these modules call `authorizeRequest` before making an HTTP call with credentials.  Turns out..  that isn't necessary at all :) The `auth.request` method in `google-auth-library` will get credentials and make the auth all in one API call for you.  If you look further down, we invoke `callback` as part of the failed promise rejection, so it's still getting invoked :)